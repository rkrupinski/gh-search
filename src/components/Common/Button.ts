import styled from "styled-components";
import { darken } from "polished";

export const Button = styled.button`
  box-sizing: border-box;
  height: 2rem;
  padding: 0 0.75rem;
  border: 0;
  border-radius: ${({ theme }) => theme.layout.borderRadius};
  font-size: 0.875rem;
  line-height: 2;
  color: ${({ theme }) => theme.colors.textAlt};
  background: ${({ theme }) => theme.colors.primary};
  appearance: none;
  transition: opacity ${({ theme }) => theme.animation.duration};

  &:disabled {
    opacity: 0.7;
  }

  &:not(:disabled) {
    cursor: pointer;

    :hover {
      background: ${({ theme }) => darken(0.1, theme.colors.primary)};
    }
  }
`;
