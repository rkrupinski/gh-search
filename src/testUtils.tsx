import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { StoreProvider } from "store";
import { theme } from "theme";

const Providers: React.FC = ({ children }) => (
  <StoreProvider>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </StoreProvider>
);

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries" | "wrapper">,
) => render(ui, { wrapper: Providers, ...options });

export const noop = () => {
  /**/
};
