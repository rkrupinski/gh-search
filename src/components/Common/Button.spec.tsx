import React from "react";
import { fireEvent } from "@testing-library/react";
import { Button } from "./Button";
import { renderWithProviders } from "testUtils";

describe("Button", () => {
  it("should render correctly", () => {
    const { container } = renderWithProviders(<Button type="submit">Button text</Button>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should let one handle click", () => {
    const handleClick = jest.fn();

    const { container } = renderWithProviders(<Button onClick={handleClick}>Button text</Button>);

    fireEvent.click(container.firstChild!);

    expect(handleClick).toHaveBeenCalled();
  });

  it("should not handle clicks when disabled", () => {
    const handleClick = jest.fn();

    const { container } = renderWithProviders(
      <Button onClick={handleClick} disabled>
        Button text
      </Button>,
    );

    fireEvent.click(container.firstChild!);

    expect(handleClick).not.toHaveBeenCalled();
  });
});
