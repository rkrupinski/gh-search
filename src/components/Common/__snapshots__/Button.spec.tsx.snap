// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
.c0 {
  box-sizing: border-box;
  height: 2rem;
  padding: 0 0.75rem;
  border: 0;
  border-radius: .5rem;
  font-size: 0.875rem;
  line-height: 2;
  color: #fff;
  background: #452cdc;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: opacity 50ms;
  transition: opacity 50ms;
}

.c0:disabled {
  opacity: 0.7;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:not(:disabled):hover {
  background: #341eb7;
}

<button
  class="c0"
  type="submit"
>
  Button text
</button>
`;
